flumine/markets/middleware.py

```diff
@@ -99,7 +99,40 @@ def _process_runner_removal(
                         extra=order.info,
                     )
                 else:
+                    # TODO: "Where an SP lay bet in a win market has a maximum odds limit specified,..."
                     if (
+                        order.order_type.ORDER_TYPE == OrderTypes.MARKET_ON_CLOSE
+                    ) and order.side == "LAY":
+                        if market.market_type == "WIN":
+                            runner = [
+                                x
+                                for x in market.market_book.runners
+                                if x.selection_id == order.selection_id
+                                and x.handicap == order.handicap
+                            ][0]
+                            runner_adjustment_factor = runner.adjustment_factor
+                            # See https://github.com/liampauling/flumine/issues/454
+                            multiplier = 1 - (
+                                removal_adjustment_factor
+                                / (100 - runner_adjustment_factor)
+                            )
+                            order.order_type.liability *= multiplier
+                            logger.warning(
+                                "WIN MARKET_ON_CLOSE Order adjusted due to non runner {0}".format(
+                                    order.selection_id
+                                ),
+                                extra=order.info,
+                            )
+                        elif market.market_type in {"PLACE", "OTHER_PLACE"}:
```

Comment:
Each way not included?

Reply:
The rules page does not mentioned each way: https://www.betfair.com/aboutUs/Rules.and.Regulations/

Also, eyeballing a handful of eachway markets, I can't find any that offer BSP. Does BSP exist on each way markets?